README.md

```diff
@@ -27,30 +29,139 @@ npm install @dojo/widgets
+```
+
+Because each widget is a separate module, when you create a release build of your application, you will only include the widgets that you have explicitly imported.
+This allows our build tooling ([`dojo cli`](https://github.com/dojo/cli)), to make sure that the production build of your application only includes the widgets you use and is as small as possible. 
 
 ## Features
 
-Dojo 2 widgets are supported in all evergreen browsers (IE11+, Chrome, Firefox, Safari) as well as popular mobile browsers (Mobile Safari, Chrome on Android).
+- All widgets are supported in all evergreen browsers (Chrome, Edge, Firefox, IE11+ and Safari) as well as popular mobile browsers (Mobile Safari, Chrome on Android).
+
+- All widgets support accessibility (`a11y`) out of the box.
+
+- All widgets are fully themeable.
+In a future release we will ship with several official Dojo 2 themes that can be applied to the widgets.
+
+- All widgets support internationalization (`i18n`)
+
+## Widgets
+
+### Form widgets
+[Button](src/button/README.md)
+
+[Checkbox/Switch](src/checkbox/README.md)
 
-Dojo 2 widgets are fully accessible (`a11y`), out of the box.
+[ComboBox](src/combobox/README.md)
 
-Dojo 2 widgets are fully themeable. In a future release we will ship with several official Dojo 2 themes that can be applied to the widgets (all themes will live in a separate repo).
+[Label](src/label/README.md)
 
-Dojo 2 widgets are fully internationalisable (`i18n`)
+[Radio](src/radio/README.md)
+
+[Select](src/select/README.md)
+
+[Slider](src/slider/README.md)
+
+[TextArea](src/TextArea/README.md)
+
+[TextInput](src/TextInput/README.md)
+
+### Layout widgets
+[SlidePane](src/slidepane/README.md)
+
+[TabPane](src/tabpane/README.md)
+
+### Misc widgets
+[Dialog](src/dialog/README.md)
 
 ## Conventions
 
+### EventHandlers
+
+You can register event handlers that get called when the corresponding events occur, by passing the handlers into a widget, in their `properties`.
+The naming convention for event handlers is as follows:
+
+- if the parent of the widget has the power to decide *if* an event is successful, i.e. can cancel the event, then the child widget will call an event handler in the following format:
+
+`onRequest[X]`, e.g. for a `close` event, the event handler called by the child widget must be called `onRequestClose`
+
+Here the child widget is requesting that the `close` event take place.
+
+- for events that will occur regardless of child/parent interaction, then the `Request` naming convention is dropped:
+
+`on[X]`, e.g. for a `dismiss` event, then event handler called by the child widget must be called `onDismiss`
+
+### Controlled vs. uncontrolled form widgets
+Currently, all of our *form* widgets are controlled widgets.
+This means that a parent widget is responsible for passing in the current value of the child widget.
+
+e.g.
+
+```ts
+[in WidgetFoo.ts]
+
+w(TextInput, {value: 'Hello Dojo!'}, []);
```

Comment:
Should we also show an `onChange` property that updates the input's value in WidgetFoo's state? I know it would make the example more verbose, but to me, responding to some handler and updating parent state is key when explaining controlled widgets.